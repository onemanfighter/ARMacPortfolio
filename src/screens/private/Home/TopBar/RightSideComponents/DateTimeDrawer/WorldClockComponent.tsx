import { Box } from '@chakra-ui/react';
import Clock from 'react-clock';
import 'react-clock/dist/Clock.css';
import { uiStore, dateTimeSelector } from '@uiStore';

const ClockComponent = ({ timeZone }: { timeZone?: string }) => {
  const { dateObject } = uiStore(dateTimeSelector);

  return (
    <Box bgSize={72} borderRadius={'full'} bg={'white'}>
      <Clock size={72} value={dateObject} />
    </Box>
  );
};

const WorldClockComponent = () => {
  return (
    <Box
      display={'flex'}
      flexDir={'row'}
      justifyContent={'space-between'}
      bg={'#1f1f1f'}
      px={2}
      my={2}
      py={4}
      shadow={'0 0 10px rgba(0, 0, 0, 0.2)'}
      borderRadius={10}
    >
      <ClockComponent />
      <ClockComponent />
      <ClockComponent />
      <ClockComponent />
    </Box>
  );
};

export default WorldClockComponent;
