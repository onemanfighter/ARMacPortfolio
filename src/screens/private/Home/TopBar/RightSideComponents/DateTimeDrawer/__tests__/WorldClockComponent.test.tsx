import { render } from '@testing-library/react';
import WorldClockComponent from '../WorldClockComponent';

describe('World clock component', () => {
  it('should render correctly', () => {
    const { container } = render(<WorldClockComponent />);

    expect(container).toMatchSnapshot();
  });
});
