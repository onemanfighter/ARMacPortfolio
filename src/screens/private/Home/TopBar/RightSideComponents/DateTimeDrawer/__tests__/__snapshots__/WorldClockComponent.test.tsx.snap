// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`World clock component should render correctly 1`] = `
<div>
  <div
    class="css-vydr3r"
  >
    <div
      class="css-1xxms50"
    >
      <div
        aria-label="mock-clock"
      />
    </div>
    <div
      class="css-1xxms50"
    >
      <div
        aria-label="mock-clock"
      />
    </div>
    <div
      class="css-1xxms50"
    >
      <div
        aria-label="mock-clock"
      />
    </div>
    <div
      class="css-1xxms50"
    >
      <div
        aria-label="mock-clock"
      />
    </div>
  </div>
</div>
`;
